### PR TITLE
Handle dscp qos configuration.

### PR DESCRIPTION
**Pull request by ansao-aci**

### CODE REVIEW

**Comment by tomflynn** [comment]
Kiran: Please take a look

Please add UTs for the new functionality. For example, code coverage in QosManager.cpp dropped from ~77% to 63% with this change. We're targeting 80% for all functionality

File: agent-ovs/lib/QosManager.cpp [hunk 1]
```diff
@@ -303,6 +339,20 @@ namespace opflexagent {
...
+    void QosManager::updateQosConfigState(const shared_ptr<modelgbp::qos::DscpMarking>& qosconfig) {
+        lock_guard<recursive_mutex> guard(opflexagent::QosManager::qos_mutex);
+
+        string dscpMarking("QosDscpMarking/");
+        string dscpMarkingUri(qosconfig->getURI().toString());
+        dscpMarkingUri.erase(dscpMarkingUri.length()-dscpMarking.size());
+
+        URI reqUri(dscpMarkingUri);
+        reqToDscp.erase(reqUri);
+        uint8_t dscp = qosconfig->getMark().get();
```
Comment (shastrinator) [inline agent-ovs/lib/QosManager.cpp:351]: Do you need to check whether Mark is set on qosConfig before doing the update?

**Review by shastrinator — COMMENTED** [review]
Please add UT as Tom mentioned

File: agent-ovs/lib/test/QosManager_test.cpp [hunk 2]
```diff
@@ -93,7 +101,13 @@ static bool checkQosBurst(boost::optional<shared_ptr<QosConfigState>> pQos,
...
     return pQos.get()->getBurst() == pQosCfg->getBurst();
 }
-
+static bool checkQosDscpMarking(uint8_t dscpMarking,
+                                   shared_ptr<qos::DscpMarking>& pQosDscpMarking) {
+    if (!pQosDscpMarking)
+        return false;
+    LOG(DEBUG) << "checkQosDscpMarking " << dscpMarking;
+    return dscpMarking == pQosDscpMarking->getMark().get();
+}
```
Comment (shastrinator) [inline agent-ovs/lib/test/QosManager_test.cpp:110]: Can you add update/remove case as well?

File: agent-ovs/ovs/AccessFlowManager.cpp [hunk 3]
```diff
@@ -521,6 +529,46 @@ void AccessFlowManager::handleEndpointUpdate(const string& uuid) {
     switchManager.writeFlow(uuid, GROUP_MAP_TABLE_ID, el);
 }
 
+void AccessFlowManager::handleDscpQosUpdate(const string& interface) {
+    string objIdV4 = interface + string("ipv4");
+    string objIdV6 = interface + string("ipv6");
```
Comment (shastrinator) [inline agent-ovs/ovs/AccessFlowManager.cpp:534]: Please add tests in AccessFlowManager_test.cpp for these flows

File: agent-ovs/lib/QosManager.cpp (hunk 1 above)
Comment (ansao-aci) [reply agent-ovs/lib/QosManager.cpp:354]: Added the check before calling updateQosConfigState.

File: agent-ovs/ovs/AccessFlowManager.cpp (hunk 3 above)
Comment (ansao-aci) [reply agent-ovs/ovs/AccessFlowManager.cpp:534]: done.

File: agent-ovs/lib/test/QosManager_test.cpp (hunk 2 above)
Comment (ansao-aci) [reply agent-ovs/lib/test/QosManager_test.cpp:110]: done

File: agent-ovs/lib/QosManager.cpp (hunk 1 above)
Comment (ansao-aci) [reply agent-ovs/lib/QosManager.cpp:354]: done

**Comment by ansao-aci** [comment]
Build is failing with these logs.
$ git clone --depth=50 https://github.com/noironetworks/opflex.git noironetworks/opflex
1216Cloning into 'noironetworks/opflex'...
1217remote: Enumerating objects: 1767, done.
1218remote: Counting objects: 100% (1767/1767), done.
1219remote: Compressing objects: 100% (1116/1116), done.
1220remote: Total 1767 (delta 821), reused 982 (delta 530), pack-reused 0
1221Receiving objects: 100% (1767/1767), 13.94 MiB | 12.01 MiB/s, done.
1222Resolving deltas: 100% (821/821), done.
1223$ cd noironetworks/opflex
12244.07s$ git fetch origin +refs/pull/252/merge:
1225fatal: the remote end hung up unexpectedly
1226
1227The command "eval git fetch origin +refs/pull/252/merge: " failed. Retrying, 2 of 3.
1228
1229fatal: the remote end hung up unexpectedly
1230
1231The command "eval git fetch origin +refs/pull/252/merge: " failed. Retrying, 3 of 3.
1232
1233fatal: the remote end hung up unexpectedly
1234
1235The command "eval git fetch origin +refs/pull/252/merge: " failed 3 times.
1236
1237The command "git fetch origin +refs/pull/252/merge:" failed and exited with 128 during .
1238
1239Your build has been stopped.

**Comment by shastrinator** [comment]
Maybe a transient issue. Can you try resubmitting with a minor change?

**Comment by ansao-aci** [comment]
Hitting the  same build error ("fatal: the remote end hung up unexpectedly") again.

**Comment by shastrinator** [comment]
Same pull seems to work fine when I try. Please create a different branch, copy the changes over and submit a new PR.

**Comment by shastrinator** [comment]
https://travis-ci.com/github/noironetworks/opflex/jobs/391214577

**Comment by ansao-aci** [comment]
Closing this due to build error. Raised a new pr from a different branch.

**Pull request closed without merging**